.github/workflows/tests.yaml

```diff
@@ -13,6 +13,47 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/main' && !startsWith(github.ref, 'refs/heads/release/') }}
 
 jobs:
+  signoffs:
+    name: Sign-offs
+    runs-on: ${{ vars.GHA_WORKER_SMALL || 'ubuntu-latest' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check all branch commits are signed off
+        if: github.event_name != 'pull_request'
+        run: |
+          for c in $(git rev-list --no-merges ${{ github.event.before }}..${{ github.event.after }}); do
+            if ! git log -1 --pretty=%B "$c" | grep -q '^Signed-off-by:'; then
+              echo "::error ::Commit $c has not been signed off in the commit message with a \`Signed-off-by:\` line"
```

Comment:
Should we add a link to https://github.com/nats-io/nats-server/blob/main/CONTRIBUTING.md#sign-off to the error message? When someone sees this message it may be not 100% what to do to fix it, and they will come with questions...

Reply:
We do already have a link to CONTRIBUTING in the PR template.